### ISSUE TITLE
STRM file not queued in current plalist

### ISSUE THREAD

**Issue by elbowz** [xbmc/xbmc#16597]
### Describe the bug
Strm file could not be added in the "current playlist" by "Queue item" (context menu).
I have "discover" this bug using the Netflix plugin, after reporting the [issue to developer](https://github.com/CastagnaIT/plugin.video.netflix/issues/199) of plugin.

## Expected Behavior
The strm is added to current playlist (left side menu "go to playlist")

## Actual Behavior

The strm don't appear in playlist...but works if you use "play from here"

## Possible Fix
See what do "play form here" and adapt to "queue item" ;)

### To Reproduce
* Install youtube plugin
* Select a *strm file* 
* click `Q` or "context menu > queue item"
* Should be appear in "current playlist" (left side menu "go to playlist")

I have used two strm file with these content:
test1.strm
```
plugin://plugin.video.youtube/play/?video_id=X79qX_kAloY
```
test2.strm
```
plugin://plugin.video.youtube/play/?video_id=jdeJUu6um0I
```

### Debuglog

The debuglog can be found here: https://pastebin.com/Jh5S7njQ

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [x] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: 
Linux 5.2.11-arch1-1-ARCH #1 SMP PREEMPT Thu Aug 29 08:09:36 UTC 2019 x86_64 GNU/Linux
 - Kodi version: 
18.3-1

Thanks for your time and hard works!

**Comment by DaveTBlake** [comment]
@elbowz please try with v18.4, it is likely this issue has been fixed and I notice from the log (thanks for providing one)

**Comment by elbowz** [comment]
Hi @DaveTBlake,

I have tried with kodi 18.4, on different Linux machine, but the behavior is the same:
Nothing is added to the current playlist.

Here the new log; https://pastebin.com/3vv1Q6ZV

Thanks for your time and suggestions

**Comment by DaveTBlake** [comment]
OK, so a different STRM file issue. I can confim that I can repeat this issue on v18.4 so I will take a look.

**Comment by elbowz** [comment]
I cannot do further test for next next 8 days...so take your times :)

Thanks @DaveTBlake for your time!

**Comment by DaveTBlake** [comment]
Looks like it was a simple fix, if you can confirm @elbowz 
(Although the youtube addon is broken at the moment because of changes by Youtube which means that testing needs to use some other plugin)

**Comment by elbowz** [comment]
Sorry, as I have written I cannot do test for the next 8 days.

So on 19th where I have to download the fixed version?!



P.s. i'll test it with Netflix plugin

**Comment by DaveTBlake** [comment]
This fix can be tested in the current v19 nightly builds, feedback appreciated and necessaru before it will be backported to v18. I can create a v18 test build for some platforms if needed, but not a ppa so not sure how much help that would be to you.

**Comment by DaveTBlake** [comment]
@elbowz are you alble to test this, or what do you need from me?

**Comment by elbowz** [comment]
I'm compiling v19 nightly build, maybe 2-4 hours and I'll make you know if works.

**Comment by elbowz** [comment]
> @elbowz are you alble to test this, or what do you need from me?

Sorry for the delay, but I have got some trouble during compilation.

Anyway, on *Kodi 19.0-ALPHA1 Git:20190924-3be901563bb* the strm file was added to the current playlist! 

So, IT'S WORKS!!

...but, there is a last issue :)

When I add to queue the strm files (Youtube or Netflix plugin) the name displayed in the Playlist is taken from last part of url in the strm file.

eg.

| STRM content | playlist name |
| ------------------- | ------------------ |
| plugin://plugin.video.youtube/play/?video_id=X79qX_kAloY | play |
| plugin://plugin.video.netflix/play/movie/80210932/ | 80210932 |

I don't know if is related or it's want you want, but with this label it's impossible understand what is the video content by reading the playlist elements (ie. for all youtube video you'll see *"play"*)

Thanks for all your effort! :+1:

**Comment by DaveTBlake** [comment]
Thanks for testing the queuing issue @elbowz 

I'm not sure about the naming problem, could be the addons responsibility to give item nice names. Is it only a queued item issue, and is it new to v19?

**Comment by elbowz** [comment]
I don't know if it's new to v19, on v18 the queue list don't works (so I cannot test).

I don't think the guilties are the plugins. 
I have test with Youtube and Netflix plugins, in `Library mode` and `Files mode` (ie. simple add strm file to queue, bypassing plugin view), but the result is the same I have reported in my previous post.

Anyway, you can try yourself...maybe it is the desiderate behaviour. 

This an example of what I get:
![Screenshot_20190926_172524](https://user-images.githubusercontent.com/1451484/65702193-e262b200-e082-11e9-9d3b-408852ce1ab1.png)

IMHO the right behaviour should be add `filename` or `displayed name` (in Library mode) to the current playlist queue. 

Maybe I missing something....

**Comment by DaveTBlake** [comment]
I am probably missing something (I don't use addons or even video very much)...
I can see the issue, what appears in the current playlist queue isn't very useful, but since it is just a strm file containing a url that is passed to the player there is no way for it to know the and display the filename. 

If there is a playback route that does display that information for a strm file containing a plugin url then I can check how that works, but I'm in the dark about this

**Comment by elbowz** [comment]
STRM file is not only for plugins, you can also write a simple url inside. 
For example, if you have a file with this content:  
`https://www.soundhelix.com/examples/mp3/SoundHelix-Song-1.mp3`  
you'll obtain `SoundHelix-Song-1.mp3` in the current playlist...and this could be reasonable. 
But today, the url is not more a `simple remote path` but a complex routing path with `get parameters`.

I guess this could be a suggestion or a request feature more then a bug :)

TL;TR for me the bug is fixed and the issue can be closed. Thanks! :) 
if there is same place where you (or me) can post this suggestion cold be great.

**Comment by DaveTBlake** [comment]
Fixed by #16604
